### ISSUE TITLE
Why does training get stuck by ipdb ?

### ISSUE THREAD

**Issue by MenSanYan** [chenxinpeng/ARNet#1]
Hi! Thank you for your sharing your code and a comprehensive tutorial to use it. However, after I excuted the command  "./bash_image_caption_soft_att_xe.sh", some weird things happened as follows:
 "
  idx: 13024  epoch: 0  lr:0.00050000  loss: 43.086  time: 0.239
  idx: 13040  epoch: 0  lr:0.00050000  loss: 41.745  time: 0.287
  > /home/xxx/extend/ARNet/image_captioning/utils_model.py(162)forward()
      161 
  --> 162         output = - input.gather(1, target) * mask
      163         output = torch.sum(output) / batch_size

  ipdb> list
      157 
      158         target_cpu = target.data.cpu().numpy()
      159         if 10516 in target_cpu:
      160             ipdb.set_trace()
      161 
  --> 162         output = - input.gather(1, target) * mask
      163         output = torch.sum(output) / batch_size
      164 
      165         return output
      166 
      167 
"
What's wrong with it ? Thank you for your answering.

**Comment by chenxinpeng** [comment]
Hi,

Thank you for reporting this issue.

What's the version of your PyTorch? This code should be evaluated on PyTorch 0.12.

**Comment by chenxinpeng** [comment]
Hi, @MenSanYan

Thank you for reporting this issue.

What's the version of your PyTorch? This code should be evaluated on PyTorch 0.12.

**Comment by chenxinpeng** [comment]
@MenSanYan 
If you want to run the code on the other versions of PyTorch, such as 0.4 or higher, please generating  the vocabulary by yourself.

**Comment by MenSanYan** [comment]
The version of my PyTorch is 0.4.
I can't find PyTorch 0.12 ( only  PyTorch 0.1.12  is found on the official website ). [https://pytorch.org/docs/versions.html](url)
I tried PyTorch 0.1.12, but the same problem happened again.

After the vocabulary is generated, all the words are indexed from 1 to 10516.
In class_soft_att.py (line 42 and line 43):
`self.embed = nn.Embedding(self.vocab_size + 1, self.input_encoding_size)
self.logit = nn.Linear(self.lstm_size, self.vocab_size)`
I see that the embedding matrix contains one more entry than the vocabulary (from 0 to 10516, entry 0 indicates a dummy word) and the ouput size is the same as the vocabulary (from 0 to 10515).
Are the predicted words shifted one position? I can't understand it.
Thank you for your answering.

**Comment by MenSanYan** [comment]
The pretrained Attentive Encoder-Decoder weights downloaded is used for inference, however, the results are not the same as in the published paper (BLEU scores are a little lower). Why is that?

 ratio: 1.0031549103621538
Bleu_1: 0.724  <  _0.727(published)_
Bleu_2: 0.554  <  _0.557(published)_
Bleu_3: 0.418  <  _0.421(published)_
Bleu_4: 0.316  <  _0.318(published)_
computing METEOR score...
METEOR: 0.257 <  _0.259(published)_
computing Rouge score...
ROUGE_L: 0.536  > _0.537(published)_
computing CIDEr score...
CIDEr: 0.996  =  _0.996(published)_
computing SPICE score...
SPICE: 0.185  = _0.185(published)_

**Comment by chenxinpeng** [comment]
@MenSanYan 
Hi,
I evaluate the model just now again, the scores are illustrated as follows:  ![scores](https://ws1.sinaimg.cn/large/006tKfTcgy1fs86r28k9rj31g00geafx.jpg)

The original records is here:
![scores](https://ws3.sinaimg.cn/large/006tKfTcgy1fs86z5s5lfj31kw01v0ta.jpg)

Please check the code carefully.  There may be some difference.

**Comment by chenxinpeng** [comment]
@MenSanYan 
Hi,

Yes, the version of PyTorch for this code is 0.1.12.

If you want to run the code with PyTorch on higher version, please change the `self.logit` as follows:
```python
self.logit = nn.Linear(self.lstm_size, self.vocab_size + 1)
```

I have tested it already.

**Comment by MenSanYan** [comment]
@chenxinpeng 
Truly sorry to bother you.
Could you accept me as your qq friend ? My qq number is 2271405258.
能qq上请教您几个问题吗？您的qq号码是？ 
Thank you very much.

**Comment by MenSanYan** [comment]
@chenxinpeng 
I uninstalled PyTorch 0.4 and installed PyTorch 0.1.12
Finally I get the same results as published!!!
ratio: 1.0000644523696773
Bleu_1: 0.727
Bleu_2: 0.557
Bleu_3: 0.421
Bleu_4: 0.318
computing METEOR score...
METEOR: 0.256
computing Rouge score...
ROUGE_L: 0.537
computing CIDEr score...
CIDEr: 0.996
computing SPICE score...
Parsing reference captions
Parsing test captions
SPICE evaluation took: 11.69 s
SPICE: 0.185
How weird! Incredible.
Thank you.
